src/components/authorization/authorization.service.ts

```diff
@@ -0,0 +1,662 @@
+    await this.db
+      .query()
+      .match([
+        [
+          node('sg', 'SecurityGroup', {
+            id: request.sgId,
+          }),
+          relation('out', '', 'member', {
+            admin: true,
+          }),
+          node('requestingUser', 'User'),
+          relation('out', '', 'token', {
+            active: true,
+          }),
+          node('token', 'Token', {
+            active: true,
+            value: session.token,
+          }),
+        ],
+        [
+          node('user', 'User', {
+            id: request.userId,
+          }),
+          relation('in', 'm', 'member'),
+          node('sg'),
+        ],
+      ])
+      .set({
+        values: {
+          'm.admin': true,
+        },
+      })
+      .first();
+
+    const result = await this.getSecurityGroupMember(
+      request.sgId,
+      request.userId
+    );
+
+    if (!result || !result.sgId || !result.userId || !result.admin) {
+      return false;
+    }
+
+    return true;
+  }
+
+  async promoteUserToAdminOfBaseNode(
+    request: PromoteUserToAdminOfBaseNode,
+    session: ISession
+  ): Promise<boolean> {
+    /**
+     * In order to be able to promote user to an admin on a base node,
+     * the requesting user must be an admin on the base node
+     */
+    await this.db
+      .query()
+      .match([
+        [
+          node('token', 'Token', {
+            active: true,
+            value: session.token,
+          }),
+          relation('in', '', 'token', {
+            active: true,
+          }),
+          node('requestingUser', 'User'),
+          relation('in', '', 'admin', {
+            active: true,
+          }),
+          node('accessBaseNode', 'BaseNode', {
+            id: request.baseNodeId,
+          }),
+        ],
+        [
+          node('user', 'User', {
+            id: request.userId,
+          }),
+        ],
+        [
+          node('baseNode', 'BaseNode', {
+            id: request.baseNodeId,
+          }),
+        ],
+      ])
+      .merge([
+        node('user'),
+        relation('in', '', 'admin', {
+          active: true,
+        }),
+        node('baseNode'),
+      ])
+      .run();
+
+    const result = await this.db
+      .query()
+      .match([
+        [
+          node('user', 'User', {
+            id: request.userId,
+          }),
+          relation('in', 'a', 'admin', {
+            active: true,
+          }),
+          node('baseNode', 'BaseNode', {
+            id: request.baseNodeId,
+          }),
+        ],
+      ])
+      .return({
+        user: [{ id: 'userId' }],
+        baseNode: [{ id: 'baseNodeId' }],
+        a: [{ active: 'adminActive' }],
+      })
+      .first();
+
+    if (
+      !result ||
+      !result.adminActive ||
+      result.userId !== request.userId ||
+      result.baseNodeId !== request.baseNodeId
+    ) {
+      return false;
+    }
+
+    return true;
+  }
+
+  async deleteSecurityGroup(id: string, session: ISession): Promise<void> {
```

Comment:
The delete function doesn't check to see if the requesting user is an admin of the security group.

Reply:
Done